### PR TITLE
Add era-specific integration test assertions

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

ADP-681

# Overview

- [x] Add a `_mainEra :: ApiEra` to the test `Context`
- [x] GET /network/parameters `eras` object should be non-Nothing for `byron .. mainEra`, and Nothing for the rest.
- [x] GET /network/information `node_era` `shouldBe` `mainEra`
- [x] Remove redundant eventually

# Comments

- [x] Do these tests _actually_ cover a faulty implementation? -> Yes (if run manually on all eras)
    - When replacing `QueryIfCurrentMary` with `QueryIfCurrentAllegra`
        - `LOCAL_CLUSTER_ERA=mary` -> uncaught
        - `LOCAL_CLUSTER_ERA=allegra` caught (error below)
        - `LOCAL_CLUSTER_ERA=shelley` uncaught 

```
       For era: ApiMary
       While verifying (Status {statusCode = 200, statusMessage = "OK"},Right (ApiNetworkParameters {genesisBlockHash = ApiT {getApiT = Hash "G\204\242\&0\STXQ\253D\STX&9\254\200\134\230\207\207\222\DC1\206\177B\201\175>\SUB\b\244\EMr/?"}, blockchainStartTime = ApiT {getApiT = StartTime 2021-02-01 13:07:11 UTC}, slotLength = Quantity {getQuantity = 0.2s}, epochLength = Quantity {getQuantity = 50}, securityParameter = Quantity {getQuantity = 5}, activeSlotCoefficient = Quantity {getQuantity = 50.0}, decentralizationLevel = Quantity {getQuantity = Percentage {getPercentage = 3 % 4}}, desiredPoolNumber = 3, minimumUtxoValue = Quantity {getQuantity = 1000000}, eras = ApiEraInfo {byron = Just (ApiEpochInfo {epochNumber = ApiT {getApiT = EpochNo {unEpochNo = 0}}, epochStartTime = 2021-02-01 13:07:11 UTC}), shelley = Just (ApiEpochInfo {epochNumber = ApiT {getApiT = EpochNo {unEpochNo = 0}}, epochStartTime = 2021-02-01 13:07:11 UTC}), allegra = Just (ApiEpochInfo {epochNumber = ApiT {getApiT = EpochNo {unEpochNo = 0}}, epochStartTime = 2021-02-01 13:07:11 UTC}), mary = Just (ApiEpochInfo {epochNumber = ApiT {getApiT = EpochNo {unEpochNo = 0}}, epochStartTime = 2021-02-01 13:07:11 UTC})}}))
       expected: Nothing
        but got: Just (ApiEpochInfo {epochNumber = ApiT {getApiT = EpochNo {unEpochNo = 0}}, epochStartTime = 2021-02-01 13:07:11 UTC})
```

### CODE REVIEW

File: lib/shelley/src/Cardano/Wallet/Shelley/Launch/Cluster.hs [hunk 1]
```diff
@@ -403,10 +405,10 @@ poolConfigsFromEnv = isEnvSet "NO_POOLS" <&> \case
     False -> defaultPoolConfigs
     True -> []
 
-localClusterConfigFromEnv :: Maybe ClusterEra -> IO LocalClusterConfig
-localClusterConfigFromEnv defaultEra = LocalClusterConfig
+localClusterConfigFromEnv :: IO LocalClusterConfig
+localClusterConfigFromEnv = LocalClusterConfig
     <$> poolConfigsFromEnv
-    <*> clusterEraFromEnv defaultEra
```
Comment (Anviking) [inline lib/shelley/src/Cardano/Wallet/Shelley/Launch/Cluster.hs:409]: I think this was intended for running the integration tests on multiple eras on the same run.

Now I want to retrieve the `clusterEra` independently.

Arguably, a `localClusterConfigFromEnv :: ClusterEra -> IO LocalClusterConfig` that doesn't use `clusterEraFromEnv` internally might be neater in that case, but 🤷‍♂️

File: lib/shelley/src/Cardano/Wallet/Shelley/Launch/Cluster.hs [hunk 2]
```diff
@@ -416,9 +418,17 @@ data ClusterEra
     | MaryHardFork
     deriving (Show, Read, Eq, Ord, Bounded, Enum)
 
+-- | Convert @ClusterEra@ to a @ApiEra@.
+clusterToApiEra :: ClusterEra -> ApiEra
+clusterToApiEra = \case
```
Comment (Anviking) [inline lib/shelley/src/Cardano/Wallet/Shelley/Launch/Cluster.hs:423]: Could replace `ClusterEra` with `ApiEra`, but I think this is fine. Maybe they diverge sometime.

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Network.hs [hunk 3]
```diff
@@ -1,4 +1,6 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MonoLocalBinds #-}
```
Comment (Anviking) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Network.hs]:
```suggestion
```

Think this is redundant. I had to add a type-signature anyway.

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Network.hs [hunk 4]
```diff
@@ -39,22 +44,40 @@ import qualified Network.HTTP.Types.Status as HTTP
...
+    it "NETWORK_PARAMS - Able to fetch network parameters" $ \ctx -> do
         r <- request @ApiNetworkParameters ctx Link.getNetworkParams Default Empty
         expectResponseCode @IO HTTP.status200 r
         let Right d = Quantity <$> mkPercentage (3 % 4)
         -- for Shelley desiredPoolNumber is node's nOpt protocol parameter
         -- in integration test setup it is 3
         let nOpt = 3
-        verify r
+
+        let clusterEra = _mainEra ctx
```
Comment (Anviking) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Network.hs]:
```suggestion
```

No point in this binding

**Review by piotr-iohk — APPROVED** [review]
lgtm

**Pull request merged**